CustomDate - copia.js

```diff
@@ -0,0 +1,127 @@
+class CustomDate {
+    constructor() {
+        this.now = new Date();
+        this.date = new Date();
+        
+        //Use constants to avoid 'magic numbers'
+        this.daysOfTheWeek = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
+        this.HoursInDay = 24;
+        this.MinsInHour = 60;
+        this.SecondsInMin = 60;
+        this.msInSecond = 1000;
+    }
+
+
+    // update this.date object
+    updateDate(newDate) {
+        this.date = newDate;
+    }
+
+
+    // return yesterday's date
+    yesterday() {
+        const yesterday = new Date();
+        yesterday.setDate(now.getDate() - 1);
+        return yesterday;
+    }
+
+
+    // return yesterday's day
+    yesterdayDay() {
+        return this.daysOfTheWeek[(this.now.getDay() - 1) % 7];
+    }
+
+
+    // return tomorrow's date
+    tomorrow() {
+        const tomorrow = new Date();
+        tomorrow.setDate(now.getDate() + 1);
+        return tomorrow;
+    }
+
+
+    // return tomorrow's day
+    tomorrowDay() {
+        return this.daysOfTheWeek[(this.now.getDay() + 1) % 7];
```

Comment:
you can do this `return this.tomorrow().toLocaleDateString(locale, { weekday: 'long' });`